src/zip.c

```diff
@@ -699,7 +699,7 @@ int zip_entry_fread(struct zip_t *zip, const char *filename) {
 
   xattr = (info.m_external_attr >> 16) & 0xFFFF;
   if (xattr > 0) {
-    if (chmod(filename, xattr) < 0) {
+    if (chmod(filename, (int)xattr) < 0) {
```

Comment:
cast to `mode_t`

Reply:
On MinGW it takes `int`

Reply:
`mode_t` is not defined on MinGW?
I hop good thing about `mode_t` is it differs per platform/compiler.

Reply:
if `mode_t` is  not defined on compiler/platform we need to define `mode_t` as an alias first.

Reply:
Replaced with `mode_t`. MinGW actually has `mode_t`. And `mode_t` doesn't produce warnings.